validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/Eth1DataCache.java

```diff
@@ -13,50 +13,48 @@
 
 package tech.pegasys.teku.validator.coordinator;
 
-import static tech.pegasys.teku.util.config.Constants.EPOCHS_PER_ETH1_VOTING_PERIOD;
-import static tech.pegasys.teku.util.config.Constants.ETH1_FOLLOW_DISTANCE;
-import static tech.pegasys.teku.util.config.Constants.SECONDS_PER_ETH1_BLOCK;
-import static tech.pegasys.teku.util.config.Constants.SECONDS_PER_SLOT;
-import static tech.pegasys.teku.util.config.Constants.SLOTS_PER_EPOCH;
-
-import com.google.common.eventbus.EventBus;
-import com.google.common.eventbus.Subscribe;
 import com.google.common.primitives.UnsignedLong;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.NavigableMap;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentSkipListMap;
+import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.datastructures.blocks.Eth1Data;
 import tech.pegasys.teku.datastructures.state.BeaconState;
-import tech.pegasys.teku.pow.event.CacheEth1BlockEvent;
-import tech.pegasys.teku.util.config.Constants;
 
 public class Eth1DataCache {
-
   private final UnsignedLong cacheDuration;
-  private volatile Optional<UnsignedLong> genesisTime = Optional.empty();
+  private final Eth1VotingPeriod eth1VotingPeriod;
 
   private final NavigableMap<UnsignedLong, Eth1Data> eth1ChainCache = new ConcurrentSkipListMap<>();
 
-  public Eth1DataCache(EventBus eventBus) {
-    eventBus.register(this);
-    cacheDuration = calculateCacheDuration();
+  public Eth1DataCache(final Eth1VotingPeriod eth1VotingPeriod) {
+    this.eth1VotingPeriod = eth1VotingPeriod;
+    cacheDuration = eth1VotingPeriod.getCacheDurationInSeconds();
   }
 
-  public void startBeaconChainMode(BeaconState headState) {
-    this.genesisTime = Optional.of(headState.getGenesis_time());
+  public void onBlockWithDeposit(final UnsignedLong blockTimestamp, final Eth1Data eth1Data) {
+    eth1ChainCache.put(blockTimestamp, eth1Data);
+    prune(blockTimestamp);
   }
 
-  @Subscribe
-  public void onCacheEth1BlockEvent(CacheEth1BlockEvent cacheEth1BlockEvent) {
-    final UnsignedLong latestBlockTimestamp = cacheEth1BlockEvent.getBlockTimestamp();
-    eth1ChainCache.put(latestBlockTimestamp, createEth1Data(cacheEth1BlockEvent));
-    prune(latestBlockTimestamp);
+  public void onEth1Block(final Bytes32 blockHash, final UnsignedLong blockTimestamp) {
+    final Entry<UnsignedLong, Eth1Data> previousBlock = eth1ChainCache.floorEntry(blockTimestamp);
+    if (previousBlock == null) {
+      // This block is either before any deposits so will never be voted for
```

Comment:
Adding a LOG.debug here might be useful, in case we're getting too many redundant blocks from `Eth1BlockFetcher`

Reply:
Done.